ADR/0028-handling-snapshotenvironmentbinding-errors.md

```diff
@@ -0,0 +1,14 @@
+# 28. Handling SnapshotEnvironmentBinding Errors
+Date: 2023-08-31
+
+## Status
+Accepted
+
+## Context
+It is currently not possible to determine whether a SnapshotEnvironmentBinding is stuck in an unrecoverable state.  This is a major problem when deciding if an ephemeral SEB needs to be cleaned up by the integration service's SnapshotEnvironmentBinding controller.  An inability to clean up errored SEBs can overload the cluster.
+
+## Decision
+The integration service has a reconciler function that cleans up errored SnapshotEnvironmentBindings with an ErrorOccured condition.  This condition is set to 'true' by default and is set to 'false' when the environment becomes available.  The integration service will that all SEBs with a 'true' ErrorOccured condition and a LastUpdateTime of more than five minutes ago are unrecoverable and can be cleaned up.
```

Comment:
suggestion: `The integration service will that` -> `The integration service will consider that`